Business Hours Block: allow folks to filter the block's content

roccotripaldi opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->

* Provides a filter allowing folks to change the output of the business hours block
* We initially hoped provide schema data within the html, but that will require some more work
* I've updated https://github.com/Automattic/wp-calypso/issues/30594 with more details


#### Testing instructions:
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Get this branch going locally with docker, or try it out on Jurassic.ninja
* In your site's post editor, add a business hours block
* Ensure the block looks good on the front end
* Try filtering the content within your theme or in a helper plugin

#### Proposed changelog entry for your changes:
<!-- Please do not leave this empty. If no changelog entry needed, state as such. -->

* none

roccotripaldi committed:
Business Hours Block:

adds support for openingHours schema: https://schema.org/openingHours

jeherve commented on extensions/blocks/business-hours/business-hours.php:
```diff
@@ -12,6 +12,65 @@
+		return sprintf(
+			'<dl class="jetpack-business-hours %s" itemscope itemtype="%s">',
+			! empty( $attributes['className'] ) ? esc_attr( $attributes['className'] ) : '',
+			$itemtype
```
Should we escape this? It should always be a URL, right?

jeherve reviewed (changes requested):
That should be a nice improvement! I only have a few remarks.

roccotripaldi committed:
addressing feedback from @jeherve

roccotripaldi commented:
Thanks for the review @jeherve ! I've addressed your feedback.

I'm going to mark this as in progress because it's not as straight forward as i initially thought.

The schema here doesn't pass for [google's structured data validator](https://search.google.com/structured-data/testing-tool/u/0/).

![screen shot 2019-02-15 at 3 33 50 pm](https://user-images.githubusercontent.com/2694219/52882891-1f52e780-3137-11e9-950a-0b394b021619.png)


- We'll definitiely need a business name, and i'll need to address that in a separate PR
- We'll also need an image!? I'm not sure why that's required.
- We should also have stuff like address and phone number, and @enejb has been working on that stuff. The trick is getting Enej's work and the work here to play nicely, and that may not be feasible.


My thoughts are if we are going to do this, we should do it well, and it may be difficult to get in for this release. What do you think? Is validation a blocker?

For the first release, maybe we can just provide a filter that allows folks to insert there own schema.org markup if they want to. For example:

```
$business_hours_content = apply_filters( 'jetpack_business_hours_content', $content, $attributes );
```

jeherve commented:
> My thoughts are if we are going to do this, we should do it well 

I agree. I think validation should be a blocker. I do like your approach about providing a filter to modify the content for now.

roccotripaldi committed:
A new approach: provide a general filter for the Business Hours content.
This still gives folks a chance to add schema data and other stuff if they want,
and allows us to ship in time for the next release.

I'm adding details to https://github.com/Automattic/wp-calypso/issues/30594
where we can add own own "officaial" schema data for the next iteration.

jeherve reviewed (approved):
This should be good to merge now, thank you! Merging.

simison commented:
What happens when search engines find partial (invalid) schema data? 🤔 Would they just ignore it all?
